### PR TITLE
fix bug in hasErrors with nested errors object

### PR DESCRIPTION
**Pull request by dagda1**
## Hey, I just made a Pull Request!

The test in the PR describes the bug well and fixes a bug in `hasErrors` that is used by `handleNext` of the `<Stepper />` component.

If `hasErrors` is given an `errors` object like this:

```ts
const errors = {
  something: {},
  otherThing: {},
  someName: {
    __errors: [
      'Accepts alphanumeric values along with _(underscore) and -(hypen) as special characters',
    ],
    addError: jest.fn(),
  },
  someOtherName: {
    __errors: ['Must start with an alphabet & not contain .(period)'],
    addError: jest.fn(),
  },
  aName: {
    __errors: [],
    addError: jest.fn(),
  },
  bName: {
    __errors: [],
    addError: jest.fn(),
  },
  cName: {
    __errors: [],
    addError: jest.fn(),
  },
};
```

The `hasErrors` function will return `false` if it encounters an empty object. 

If the first error is an empty object, it will bottom out at the first error ignoring the other objects with validation errors.

#### :heavy_check_mark: Checklist

- [X] A changeset describing the change and affected packages. ([more info](https://github.com/backstage/backstage/blob/master/CONTRIBUTING.md#creating-changesets))
- [ ] Added or updated documentation
- [X] Tests for new functionality and regression tests for bug fixes
- [ ] Screenshots attached (for UI changes)
- [X] All your commits have a `Signed-off-by` line in the message. ([more info](https://github.com/backstage/backstage/blob/master/CONTRIBUTING.md#developer-certificate-of-origin))

### CODE REVIEW

**Review by benjdlambert — APPROVED** [review]

**Comment by benjdlambert** [comment]
Nice catch! thanks! :pray:

**Comment by dagda1** [comment]
@benjdlambert I'll double-check this fix at the source, please before merging.

File: plugins/scaffolder-react/src/next/components/Stepper/utils.test.ts [hunk 1]
```diff
@@ -13,6 +13,7 @@
  * See the License for the specific language governing permissions and
  * limitations under the License.
  */
+import { type FormValidation } from './createAsyncValidators';
```
Comment (benjdlambert) [inline plugins/scaffolder-react/src/next/components/Stepper/utils.test.ts:16]: don't need this :pray:

**Comment by dagda1** [comment]
@benjdlambert good to go

**Pull request merged**